### PR TITLE
Fix const eval input tensor stride and update tests

### LINKED ISSUE #609: Runtime error  : stride mismatch
## Description

- Stride mismatch runtime error in wideresnet 

## Reproduce

```
git checkout kkannan/stride_issue
git submodule update --recursive
pytest forge/test/mlir/test_ops.py -k " stride" -svv
```

## Logs

- [wideresnet_stride_issue.log](https://github.com/user-attachments/files/17615850/xx_nov4_stride_s_2.log)

### LINKED ISSUE #673: Const eval input tensor stride mismatch
When transpose is part of the const eval graph, it causes const eval input tensor to have different stride than compiled model expects. The issue is that compiled model expects the stride that is consistent with current shape of tensor. We should align constant input stride with its shape.

### PR DESCRIPTION
**Pull request by dgolubovicTT**
Closes #673. Closes #609.
- Fix const eval input tensor stride.
- Change resnet test: Run compiled ResNet model
- Remove xfail from Llama mlp and lm_head tests since they pass now.

### CODE REVIEW

File: forge/test/mlir/resnet/test_resnet_inference.py [hunk 1]
```diff
@@ -21,8 +22,14 @@ def test_resnet_inference():
     input_image = torch.rand(1, 3, 224, 224)
 
     # Sanity run
-    generation_output = framework_model(input_image)
-    print(generation_output)
+    fw_out = framework_model(input_image)
+    # print(fw_out)
```
Comment (nvukobratTT) [inline forge/test/mlir/resnet/test_resnet_inference.py:26]: Redundant line?

File: forge/test/mlir/resnet/test_resnet_inference.py [hunk 2]
```diff
@@ -21,8 +22,14 @@ def test_resnet_inference():
...
 
     # Compile the model
     compiled_model = forge.compile(framework_model, input_image)
+    co_out = compiled_model(input_image)
+    print(co_out)
```
Comment (nvukobratTT) [inline forge/test/mlir/resnet/test_resnet_inference.py:31]: Do we need to print outputs?

File: forge/forge/tensor.py [hunk 3]
```diff
@@ -1413,6 +1423,8 @@ def get_post_const_eval_tensors(
 
     for input_name in ordered_input_names:
         # Load input constant tensors for consteval
+        if input_name == "input_1_add_18":
```
Comment (nvukobratTT) [inline forge/forge/tensor.py:1426]: Redundant block?

File: forge/forge/tensor.py [hunk 4]
```diff
@@ -1339,7 +1339,12 @@ def eval_op(op_type, inputs):
 
 
 def consteval_input(consteval_trace, name: str, inputs: Dict[str, torch.Tensor], is_forge: bool) -> torch.Tensor:
-    return consteval_tensor(consteval_trace, name, inputs, is_forge, "Forward")
+    const_eval_tensor = consteval_tensor(consteval_trace, name, inputs, is_forge, "Forward")
+    # We need to change stride of new tensor explicitly by creating new tensor with shape of const_eval_tensor,
```
Comment (nvukobratTT) [inline forge/forge/tensor.py:1343]: Can we layer this explanation? In sum, to answer the following questions in order:
1. What we're doing?
2. Why we're doing it?
3. Specifics, Notes, Comments, etc. about how we implement it for the compiler.

Explanation is good, but I think it'll be good just to structure it a bit more to be easier to follow :))

File: forge/test/mlir/resnet/test_resnet_inference.py (hunk 2 above)
Comment (dgolubovicTT) [reply forge/test/mlir/resnet/test_resnet_inference.py:31]: yeah, we don't need it...

File: forge/forge/tensor.py (hunk 4 above)
Comment (dgolubovicTT) [reply forge/forge/tensor.py]: Updated the comments.

**Review by nvukobratTT — APPROVED** [review]

**Review by pilkicTT — CHANGES_REQUESTED** [review]
Let's sync on this offline, this doesn't seem to me like the correct approach... Maybe i'm missing something...

**Comment by dgolubovicTT** [comment]
@pilkicTT I would merge this one to unblock consteval cases. I created a tech debt issue to follow up with investigation for general cases of inputs not having stride consistent with their shape

**Comment by nvukobratTT** [comment]
> @pilkicTT I would merge this one to unblock consteval cases. I created a tech debt issue to follow up with investigation for general cases of inputs not having stride consistent with their shape

Let's link the new issues here, just to have a ref :))

**Review by pilkicTT — APPROVED** [review]
Let's follow up then

**Pull request merged**